### ISSUE TITLE
Crash in ImageDownloader

### ISSUE THREAD

**Issue by bamx23** [mapbox/mapbox-navigation-ios#3067]
**Mapbox Navigation SDK version:** 2.0.0-beta.12

### Steps to reproduce

```
Crashed: com.apple.NSURLSession-delegate
0  OneTap                         0x1007e72b0 ImageDownloader.activeOperation(with:) + 72 (ImageDownloader.swift:72)
1  OneTap                         0x1007e8530 specialized ImageDownloader.urlSession(_:dataTask:didReceive:completionHandler:) + 98 (<compiler-generated>:98)
2  OneTap                         0x1007e7408 @objc ImageDownloader.urlSession(_:dataTask:didReceive:completionHandler:) + 4339856392 (<compiler-generated>:4339856392)
3  CFNetwork                      0x199c15038 _CFNetworkHTTPConnectionCacheSetLimit + 155128
```

I have no exact steps to reproduce. The crash happened during a simulated navigation session here:

https://github.com/mapbox/mapbox-navigation-ios/blob/48c312c3fd2830a08b2079a58db535dfcaa8508e/Sources/MapboxNavigation/ImageDownloader.swift#L72

due to no synchronization in this method for `activeOperation` call:

https://github.com/mapbox/mapbox-navigation-ios/blob/48c312c3fd2830a08b2079a58db535dfcaa8508e/Sources/MapboxNavigation/ImageDownloader.swift#L97-L103

All 3 `URLSessionDataDelegate` methods should have synchronization for `operations` dictionary.

cc: @truburt @1ec5 @MaximAlien

**Comment by S2Ler** [comment]
The issue is in incorrect synchronisation. I will take care of this issue.

**Comment by S2Ler** [comment]
- [ ] completion blocks called 2 times if status code is >= 400

**Comment by S2Ler** [comment]
@bamx23 The fix merged to v2.0 branch. If you can test the app with these changes would be fantastic.

**Comment by 1ec5** [comment]
Fixed in #3072 on the release-v2.0 branch.

**Comment by bamx23** [comment]
Thanks, @S2Ler. As this crash happened only once in production and I don't have a way to reproduce it I'm going to wait for the next release of Nav SDK.
